File: python/coleccionesYDiccionarios/Ejercicio11.py
'''

Realiza un programa que escoja al azar 5 palabras en español del minidiccionario
del ejercicio anterior. El programa irá pidiendo que el usuario
teclee la traducción al inglés de cada una de las palabras y comprobará si
son correctas. Al final, el programa deberá mostrar cuántas respuestas son
válidas y cuántas erróneas.

@author: Javier Lopera Jimenez
'''
import random
dic = {'gato':'cat','perro':'dog','mesa':'table','puerta':'door','ventana':'window','agua':'water','aceite':'oil'
	,'pizza':'pizza','sofa':'sofa','pelicula':'film','hola':'hello','adios':'bye','salud':'health','ganar':'win'
	,'perder':'lose','cabeza':'head','menos':'less','noche':'nicht','palabra':'word','mundo':'world'}

lista=[]

listaDesordenada=[]

aciertos=0

errores=0

for i in dic:
	lista.append(i)
	
"""

Desordenamos la lista con shuffle, despues añadimos las 5 palabras aleatorias a la nueva 
lista y por ultimo preguntamos la traduccion de estas palabras.

"""

random.shuffle(lista)

for i in range(5):
	
	listaDesordenada.append(lista.pop(0))

for i in listaDesordenada:
	
	respuesta=input("\nIntroduca la traduccion de "+i + ":\n")
	
	if(respuesta==dic.get(i)):
		aciertos+=1
	else:
		errores+=1

print("Has acertado",aciertos,"palabras")
print("Has fallado",errores,"palabras")	
	
	

File: java/ejerciciosPOO3v2/coleccionesYDiccionarios/Ejercicio11.java
package coleccionesYDiccionarios;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.Scanner;

/**
 * Realiza un programa que escoja al azar 5 palabras en español del minidiccionario
 * del ejercicio anterior. El programa irá pidiendo que el usuario
 * teclee la traducción al inglés de cada una de las palabras y comprobará si
 * son correctas. Al final, el programa deberá mostrar cuántas respuestas son
 * válidas y cuántas erróneas.
 *
 * @author d18lojij
 *
 */

public class Ejercicio11 {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		HashMap<String,String> d = new HashMap<String, String>();

		ArrayList <String> claves = new ArrayList<>();
		
		ArrayList <String> clavesAleatorias = new ArrayList<>();
		
		int aciertos = 0;
		
		int fallos = 0;
		
		d.put("casa", "house");
		d.put("coche", "car");
		d.put("llave", "key");
		d.put("rojo", "red");
		d.put("azul", "blue");
		d.put("negro", "black");
		d.put("amarillo", "yellow");
		d.put("ordenador", "computer");
		d.put("bicicleta", "bike");
		d.put("mesa", "table");
		d.put("ratón", "mouse");
		d.put("teléfono", "phone");
		d.put("piso", "flat");
		d.put("naranja", "oranje");
		d.put("limón", "lemon");
		d.put("cerdo", "porc");
		d.put("gallina", "chiken");
		d.put("chocolate", "chocolate");
		d.put("miel", "honey");

		claves.addAll(d.keySet());
		
		Collections.shuffle(claves);
		
		for(int i=0;i<5;i++) {
			clavesAleatorias.add(claves.remove(0));
		}
		
		for(String i : clavesAleatorias) {
			String respuesta = "";
			System.out.println("Introduzca la traduccion de "+ i );
			respuesta = Teclado.nextLine();
			
			if(respuesta.contains(d.get(i))) {
				aciertos++;
			}
			else {
				fallos++;
			}
		}
		
		System.out.println("Has acertado " + aciertos + " palabras.");
		
		System.out.println("Has fallado " + fallos + " palabras.");
		
	}

}
